Core/src/main/java/com/plotsquared/core/configuration/caption/load/CaptionLoader.java

```diff
@@ -202,4 +215,27 @@ private static boolean patch(final Map<String, String> map, final Locale locale)
         }
         return modified;
     }
+
+    /**
+     * Returns a function that extracts a locale from a path using the given pattern.
+     * The pattern is required to have (at least) one capturing group, as this is used to access the locale
+     * tag.The function will throw an {@link IllegalArgumentException} if the matcher doesn't match the file name
+     * of the input path. The language tag is loaded using {@link Locale#forLanguageTag(String)}.
+     *
+     * @param pattern the pattern to match and extract the language tag with.
+     * @return a function to extract a locale from a path using a pattern.
+     * @see Matcher#group(int)
+     * @see Path#getFileName()
+     */
+    public static @NonNull Function<Path, Locale> patternExtractor(final @NonNull Pattern pattern) {
```

Comment:
If this is how the pattern is used, wouldn't it make more sense to make it a `Predicate<Path>`? You can then use `pattern.asPredicate()` to pass a pattern.

Reply:
We're extracting the language tag from the path with that pattern, whereas the predicate would only tell us *if* it matches

Reply:
Ah ye my bad, didn't se it being used underneath. Then it's all good!